server/request_handler.go

```diff
@@ -0,0 +1,169 @@
+package server
+
+import (
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"github.com/flashbots/rpc-endpoint/types"
+	"github.com/flashbots/rpc-endpoint/utils"
+	"github.com/google/uuid"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// RPC request handler for a single/ batch JSON-RPC request
+type RpcRequestHandler struct {
+	respw *http.ResponseWriter
+	req   *http.Request
+
+	uid             string
+	timeStarted     time.Time
+	defaultProxyUrl string
+	relaySigningKey *ecdsa.PrivateKey
+
+	handleBatch bool // If set, batch request will be handled
+}
+
+func NewRpcRequestHandler(respw *http.ResponseWriter, req *http.Request, proxyUrl string, relaySigningKey *ecdsa.PrivateKey) *RpcRequestHandler {
+	return &RpcRequestHandler{
+		respw:           respw,
+		req:             req,
+		uid:             uuid.New().String(),
+		timeStarted:     Now(),
+		defaultProxyUrl: proxyUrl,
+		relaySigningKey: relaySigningKey,
+	}
+}
+
+func (r *RpcRequestHandler) log(format string, v ...interface{}) {
+	prefix := fmt.Sprintf("[%s] ", r.uid)
```

Comment:
Logger seems duplicated here.need to think through how to handle it nicely without duplicating

Reply:
Maybe not needed here, called only once I think

Reply:
Perhaps we should pass these log functions into the request, rather than recreating it there and passing the uid in 🤔

Reply:
Done. Introduced logger to handle it

Reply:
Maybe a generator would be even better, then we can use the log functions here, pass them into the request processing functions, and add an index to the uid for batched requests.

Reply:
Yea I was thinking about it played too. But logger is an interface and little hard to pass in the new value once the logger is created. If you have some suggestion let me know